File: annotationprocessingtesting/src/main/java/com/tickaroo/tikxml/annotationprocessing/propertyelement/EmptyStringPropertyElement.java
package com.tickaroo.tikxml.annotationprocessing.propertyelement;

import com.tickaroo.tikxml.annotation.PropertyElement;
import com.tickaroo.tikxml.annotation.Xml;

/**
 * @author Hannes Dorfmann
 */
@Xml(name = "emptyPropertyTag")
public class EmptyStringPropertyElement {
  @PropertyElement
  String empty;

  @Override public boolean equals(Object o) {
    if (this == o) return true;
    if (!(o instanceof EmptyStringPropertyElement)) return false;

    EmptyStringPropertyElement that = (EmptyStringPropertyElement) o;

    return empty != null ? empty.equals(that.empty) : that.empty == null;
  }

  @Override public int hashCode() {
    return empty != null ? empty.hashCode() : 0;
  }
}

File: annotationprocessingtesting-kt/src/main/java/com/tickaroo/tikxml/annotationprocessing/propertyelement/EmptyStringPropertyElement.kt
package com.tickaroo.tikxml.annotationprocessing.propertyelement

import com.tickaroo.tikxml.annotation.PropertyElement
import com.tickaroo.tikxml.annotation.Xml

/**
 * @author Hannes Dorfmann
 */
@Xml(name = "emptyPropertyTag")
class EmptyStringPropertyElement {
    @PropertyElement
    var empty: String? = null

    override fun equals(other: Any?): Boolean {
        if (this === other) return true
        if (other !is EmptyStringPropertyElement) return false

        val that = other as EmptyStringPropertyElement?

        return if (empty != null) empty == that!!.empty else that!!.empty == null
    }

    override fun hashCode(): Int {
        return if (empty != null) empty!!.hashCode() else 0
    }
}
